src/webserver/webserver/authentication/templates/registration/login.html

```diff
@@ -0,0 +1,236 @@
+    color: #ddd;
+    font-size: 18px;
+    text-decoration: none;
+    transition: color 0.3s;
+  }
+
+  .sidebar-nav .nav-link:hover {
+    color: #8b8b8b;
+  }
+
+  .sidebar-nav .logout-btn {
+    color: #ffffff;
+    text-decoration: none;
+    font-size: 16px;
+  }
+
+  .sidebar-nav .logout-btn:hover {
+    color: #8b8b8b;
+  }
+
+  h1, h2, h3 {
+    font-weight: 600;
+    color: #333;
+  }
+
+  p {
+    font-size: 16px;
+    line-height: 1.6;
+    color: #555;
+  }
+
+  .content-wrapper {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    margin-left: 425px; 
+    height: 100vh;
+    width: calc(100% - 425px); 
+  }
+
+  .login-container {
+    width: 100%;
+    max-width: 500px; 
+    background-color: #fff;
+    padding: 40px;
+    border-radius: 8px;
+    box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
+    text-align: center;
+  }
+
+  h2 {
+    color: #333;
+    margin-bottom: 20px;
+    font-size: 24px; 
+  }
+
+  .error-message,
+  .info-message {
+    background-color: #ffdddd;
+    color: #d8000c;
+    padding: 10px;
+    margin-bottom: 15px;
+    border-radius: 4px;
+  }
+
+  .info-message {
+    background-color: #d9edf7;
+    color: #31708f;
+  }
+
+  .form-group {
+    margin-bottom: 20px;
+  }
+
+  .form-group label {
+    font-size: 16px;
+    color: #555;
+    margin-bottom: 10px;
+    display: block;
+  }
+
+  .form-group input {
+    width: 100%;
+    padding: 15px; 
+    font-size: 18px; 
+    border: 1px solid #ccc;
+    border-radius: 6px;
+    transition: border-color 0.3s;
+  }
+
+  .form-group input:focus {
+    border-color: #696969;
+    outline: none;
+  }
+
+  .form-actions {
+    text-align: center;
+    background-color: #696969;
+  }
+
+  .form-actions .btn {
+    width: 100%;
+    padding: 15px;
+    border: none;
+    font-size: 18px;  
+    border-radius: 6px;
+    cursor: pointer;
+    transition: background-color 0.3s;
+  }
+
+  .form-actions .primary-btn {
+    background-color: #696969; 
+    color: white;  
+  }
+
+  .form-actions .primary-btn:hover {
+    background-color: #515151;  
+  }
+
+  .forgot-password-link {
+    margin-top: 20px;
+    font-size: 16px;
+  }
+
+  .forgot-password-link a {
+    color: #007bff;
+    text-decoration: none;
+  }
+
+  .forgot-password-link a:hover {
+    text-decoration: underline;
+  }
+
+  @media (max-width: 768px) {
+    .content-wrapper {
+      margin-left: 0;
+      padding-top: 30px;
+    }
+
+    .login-container {
+      width: 90%;
+      padding: 30px;
+    }
+
+    h2 {
+      font-size: 22px;
+    }
+
+    .form-group input,
+    .btn {
+      font-size: 16px;
+    }
+  }
+</style>
+
+{% extends "login_page.html" %}
+
+{% block content %}
+  <div class="content-wrapper">
+    <div class="login-container">
+      <h2>Login</h2>
+
+      {% if form.errors %}
+        <div class="error-message">
+          <p>Your username and password didn't match. Please try again.</p>
+        </div>
+      {% endif %}
+
+      {% if next %}
```

Comment:
What is this if-else chain for?

Reply:
The first if checks if there was an error during the login, mainly if the username and password do not match any credentials in the system. The second if next is triggered when a user tries to access a page that requires being logged in or requires special privileges and sends a message to the user that they cannot access it